includes/reader-revenue/woocommerce/my-account/class-my-account-ui-v1.php

```diff
@@ -0,0 +1,101 @@
+<?php
+/**
+ * Newspack "My Account" customizations v1.x.x.
+ *
+ * @package Newspack
+ */
+
+namespace Newspack;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Newspack "My Account" customizations v1.x.x.
+ */
+class My_Account_UI_V1 extends WooCommerce_My_Account {
+	/**
+	 * Initialize.
+	 *
+	 * @codeCoverageIgnore
+	 */
+	public static function init() {
+		\add_filter( 'body_class', [ __CLASS__, 'add_body_class' ] );
+		\add_filter( 'do_shortcode_tag', [ __CLASS__, 'add_newspack_ui_wrapper' ], 10, 2 );
+		\add_action( 'wp_enqueue_scripts', [ __CLASS__, 'enqueue_assets' ], 11 );
+		\add_filter( 'wc_get_template', [ __CLASS__, 'wc_get_template' ], 10, 5 );
+	}
+
+	/**
+	 * Add a body class to the My Account page.
+	 *
+	 * @param array $classes The body classes.
+	 * @return array The body classes.
+	 */
+	public static function add_body_class( $classes ) {
+		if ( function_exists( 'is_account_page' ) && \is_account_page() ) {
+			$classes[] = 'newspack-my-account';
+			$classes[] = 'newspack-my-account--v1';
+			if ( ! \is_user_logged_in() ) {
+				$classes[] = 'newspack-my-account--logged-out';
+			} else {
+				$classes[] = 'newspack-my-account--logged-in';
+			}
+		}
+		return $classes;
+	}
+
+	/**
+	 * Render a wrapper element to apply Newspack UI styles to My Account page content.
+	 *
+	 * @param string $output The output.
+	 * @param string $tag The tag.
+	 *
+	 * @return string The output.
+	 */
+	public static function add_newspack_ui_wrapper( $output, $tag ) {
+		if ( 'woocommerce_my_account' === $tag ) {
+			return '<div class="newspack-ui">' . $output . '</div>';
+		}
+		return $output;
+	}
```

Comment:
Given that the goal is to have the My Account dashboard take over the entire page, what do you think of having `newspack-ui` as a body class instead?

Reply:
I tried that, but that ends up applying Newspack UI styles to more elements than we want (header, footer, etc.). Since we only want Newspack UI styles to apply to generalized components inside the post content, this was the easiest way I could figure out to do so without relying on theme-specific hooks.

Reply:
I think once we implement the full-page takeover in a separate PR, we might be able to safely move this class name to the body

Reply:
https://github.com/Automattic/newspack-plugin/pull/3974#issuecomment-2877975641